### ISSUE TITLE
NaN encounted during finetuning on udis-d dataset based on ckpt from scratch trained on stitched MS-COCO

### ISSUE THREAD

**Issue by BoomSky0416** [nie-lang/UnsupervisedDeepImageStitching#29]
Hello, I tried to reimplement the training process based on your tutorial. At first, I trained the model from scratch on stitched MS-COCO with the parameters same as your settings. The NaN issue didn't appear. But during the finetuning stage on udis-d dataset, the NaN issue happened at around 920k iterations. Here is the loss curve for two-stage training process.
![image](https://user-images.githubusercontent.com/57380777/161011735-a532457d-7906-409c-b84e-68bfcd68a827.png)
![image](https://user-images.githubusercontent.com/57380777/161011770-2bbcdc0d-b503-493f-9321-0359457d7b79.png)
Any advice to avoid NaN?

**Comment by nie-lang** [comment]
In fact, I do not encounter this problem.

How about clipping the gradients to a smaller value in the fine-tuning stage to avoid NaN?
![1648726094(1)](https://user-images.githubusercontent.com/54256137/161044972-b335814a-6339-4ed0-97f8-5eb31aa59453.jpg)

**Comment by BoomSky0416** [comment]
OK, I will try it. Thanks!
